### PR TITLE
Correct naming of taxonomy note field (TaxHTField0 and lowercase ID)

### PR DESCRIPTION
**Pull request by moontear**
This has already been described in issue #49 which has been closed.

The note-field associated with a TaxonomyField needs to have a correct naming scheme as for example explained here: https://www.c-sharpcorner.com/UploadFile/prasham/taxonomy-field-naming-convention/

Without the correct naming, SharePoint search will not create crawled properties that would be necessary to correctly search taxonomy fields.

### CODE REVIEW

**Comment by PaoloPia** [comment]
Hi,
Thank you for your contribution. The file you are proposing to change is just a sample for the sake of explaining what can be done with the PnP Provisioning Schema together with the PnP Provisioning Engine. As such, the content of the file doesn't really matter.
Moreover, your PR targets the master branch, rather than the experimental one.
So, I will close this item. Thanks.

**Comment by moontear** [comment]
Paolo I heartily disagree. Of course this is only a sample file (that's why it is within "Samples") and it doesn't really matter if the content is X or Y.

What does matter is that there is a wrong definition of taxonomy fields in this sample for - taxonomy fields. This leads to errors and since we're in the Patterns and Practices repository we should show the correct patterns or not?

Just closing this issue is not really helpful. Should I resubmit for experimental? Do you want to implement this yourself in master?

**Comment by ruslanurban** [comment]
I agree with @moontear. **Samples should be clean!** *It is the guidance for new users!*
It had caused me time and a great deal of frustration to figure out why taxonomy modeled after the samples would not show up in the search crawled properties. I had tried a number of things to fix it: reset search indexes and re-crawled everything, forced re-indexing on lists, nothing seemed to help... until I found articles that talk about prefixing the hidden taxonomy note field with **TaxHTField**... I do not know the details how SharePoint registers the note field when it is created through the UI as even though the internal name does not start with the prefix, search crawling works, but when tax fields are provisioned using other methods it requires the prefix. Must be a bug in SharePoint Search...
Bottom line is, please **fix the samples**! And, **test** them before publishing! There is a number of issues with them. If you try to provision samples, they simply do not work.

**Comment by PaoloPia** [comment]
Hi @DennisGaida and @ruslanurban,
We just updated the taxonomy sample accordingly to the current behavior of SharePoint Online, which creates the hidden note field if it is not there. The sample should be now ok for SPO. Nevertheless, I advise to export provisioning templates from sites, rather than creating them manually at the XML level. The samples in this repository are here just for the sake of explaining the schema syntax and are not intended for an everyday usage.

**Comment by DennisGaida** [comment]
This is good, but not perfect since PnP also works on-premises and AFAIK SP2019 does not create the hidden note field automatically, but I didn't check. If SP2019 also does this - that's great! Otherwise I would love a little comment/note within the sample that an additional note field is needed for search and that this sample exists simply for the sake of explaining the schema syntax.

**Comment by VesaJuvonen** [comment]
Thanks, @DennisGaida for the input. Notice that SharePoint PnP is open-source initiative 100% dependent on the contributions from the community (customers and partners). There are no full-time MS employees in this project. In this case, @PaoloPia has used his own time away from his own business to be helpful for you, which is extremely valuable for everyone in the SharePoint community. 

If you don't like the open-source solutions what the community builds, you do not need to use them and you can build your own tooling. That's not however how we hope this would work... as that's not beneficial for anyone... let's rather be thankful on the valuable input what we are getting AND potentially also contribute on any other open questions.

Having the sample now supporting SPO is a great start. If you have time to test SP2019, we do welcome contributions. We do have already more than 700 contributors in the SharePoint open-source initiates and more is absolutely welcome. If you do not have time or willingness to contribute, rather just use the tooling, that's fine as well. Would still welcome comments being thankful for the people who are trying to help you for free by using their own time for your benefit. 

Sharing is caring!

**Comment by DennisGaida** [comment]
@VesaJuvonen thanks for the information and I am well aware of everything you said. I also did invest my own time away from my business to create this pull request we are debating.
I hope I didn't sound like I was questioning the PnP initiative or Paolo or anyone for that matter. Paolo does great work, you do great work - there is no question about this. I also did "Like" Paolos comment which is a "thanks" for me in this case and I said it was "good" and could be better. All I am doing is giving my input on this pull request and the related issue on this. I would love to get back to the pull request at hand: SPO is a great start as you said and of course I'm willing to test SP2019 and whether it automatically creates hidden note fields for taxonomy fields (maybe someone knows this without me having to test, that's what my comment was for). If it does, this issue can be closed - if it doesn't my previous comment remains valid.

**Comment by corriespondent** [comment]
We have been fighting with this issue on a **2019** environment (ows_taxid crawled property not being created), so for what it's worth, here is what I found:

Deploying the taxonomy field by itself does not create a related note field: 
- I deployed just the one field , then exported a template. While I could see other note fields from manually-created taxonomy fields, I could not see one that corresponded to my deployed field.

I had come across the suggestion to add "TaxHTField" instead of "_0" at https://www.c-sharpcorner.com/UploadFile/prasham/taxonomy-field-naming-convention/ and https://github.com/pnp/PnP-Provisioning-Schema/issues/49#issuecomment-272532305 -- however, these did not work for me UNLESS I followed this specific structure: (only showing relevant fields)

`<Field 
          Type="Note" 
          ID="{d444bf15-5e57-4d06-bf17-127bb1d88050}" 
          DisplayName="DeployedTermFieldNameTaxHTField" 
          StaticName="a2b313423b08454a80555fe75595f09d" 
          Name="a2b313423b08454a80555fe75595f09d" 
          ShowInViewForms="FALSE" 
          Required="FALSE" 
          Hidden="TRUE" 
          CanToggleHidden="TRUE" 
          />
        
        <Field
          Type="TaxonomyFieldTypeMulti" 
          ID="{a2b31342-3b08-454a-8055-5fe75595f09d}" 
          DisplayName="Deployed Term Field Name" 
          StaticName="DeployedTermFieldName" 
          Name="DeployedTermFieldName"
          ...
          />`

Notice that the ID guid from my Taxonomy Field (a2b31342-3b08-454a-8055-5fe75595f09d) is what I used for the StaticName and Name of the Note field, minus the dashes (a2b313423b08454a80555fe75595f09d). For the DisplayName, I used the same StaticName/Name from the TaxonomyFieldTypeMulti, with"TaxHTField" appended.

I've experimented a few times with using friendlier text for the StaticName/Name of the Note field. While the fields are deployed without a problem, once I try to use them and crawl, the ows_taxid_DeployedTermFieldName crawled property and corresponding owstaxIdDeployedTermFieldName managed property never gets created, unless I use the setup described above with the condensed guid. I'm wondering what I'm missing as all the other links/examples are able to.

Side note: when I exported fields that were manually created, the Note field's Static/Names did NOT match the ID of the Taxonomy field:

<Field Type="Note" DisplayName="Test_Manual_Tags_0" StaticName="**m5f11d1b406e43b39c4df7d5ea0a166b**" Name="**m5f11d1b406e43b39c4df7d5ea0a166b**" ID="{b69f4808-8648-4f53-a87d-8dec3b023f1b}" ... />
        <Field Type="TaxonomyFieldTypeMulti"  ID="{**65f11d1b-406e-43b3-9c4d-f7d5ea0a166b**}" DisplayName="Test_Manual_Tags" ... />

But when I tried deploying with a random string for the StaticName, it once again did not created the crawled property. So confused!
